core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -148,45 +179,69 @@ trait EntityDecoderInstances {
   /** Provides a mechanism to fail decoding */
   def error[F[_], T](t: Throwable)(implicit F: Effect[F]): EntityDecoder[F, T] =
     new EntityDecoder[F, T] {
-      override def decode(msg: Message[F], strict: Boolean): DecodeResult[F, T] =
+      override def decode[M[_[_]]](msg: M[F], strict: Boolean)(implicit M: Message[M, F]): DecodeResult[F, T] =
         DecodeResult(msg.body.run *> F.raiseError(t))
       override def consumes: Set[MediaRange] = Set.empty
     }
 
   implicit def binary[F[_]: Effect]: EntityDecoder[F, Chunk[Byte]] =
-    EntityDecoder.decodeBy(MediaRange.`*/*`)(collectBinary[F])
+    new EntityDecoder[F, Chunk[Byte]] {
```

Comment:
I have been awake all night on a plane and am feeling stupid.  Why is `decodeBy` no longer viable?

Reply:
Higher-Rank types. I ran into the same question and was set straight by @SystemFw. Effectively As the entity decoders functions are themselves parameterized by the `M[_[_]]` which is determined by the caller. It is impossible to construct values in a method. This is equivalent to how you can have an easy value for creating a `Show` but must invoke `new Functor[F[_]]` when creating a functor as map is itself parameterized.